### PR TITLE
Implemented callback to handle device adminState

### PR DESCRIPTION
**Pull request by jduranf**
Implemented #32 

Signed-off-by: Joan Duran <jduran@circutor.com>

### CODE REVIEW

File: update.go [hunk 1]
```diff
@@ -11,31 +11,49 @@ import (
...
-		// TODO: handle error properly
-		fmt.Fprintf(os.Stderr, "service: callbackHandler invalid request: %v\n", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		svc.lc.Error(fmt.Sprintf("Invalid callback request: %v", err))
+		return
 	}
 
-	action := cbAlert.ActionType
-	id := cbAlert.Id
+	if (cbAlert.Id == "") || (cbAlert.ActionType == "") {
```
Comment (falkerson) [inline update.go]: is this covers cases when body has follow structure:
1. {}
2. {Id: null}

?

Comment (falkerson) [reply update.go]: Could you please add unit tests for it?

File: update.go [hunk 2]
```diff
@@ -11,31 +11,49 @@ import (
...
+		http.Error(w, "Missing parameters", http.StatusBadRequest)
+		svc.lc.Error(fmt.Sprintf("Missing callback parameters"))
+		return
+	}
 
-	fmt.Fprintf(os.Stderr, "service: callbackHandler action: %v id: %s\n", action, id)
+	// It was decided at the last F2F, that the one Core Metadata callback
+	// function to be supported for Dehli is handling changes to a device's
+	// adminState (LOCKED or UNLOCKED).
+	if (cbAlert.ActionType == models.DEVICE) && (req.Method == http.MethodPut) {
```
Comment (falkerson) [inline update.go]: Same here, need to cover 3 cases:
1. Valid data
2. Couldn't update device
3. Invalid data

File: update.go (hunk 1 above)
Comment (jduranf) [reply update.go]: Yes, covers those cases

File: update.go (hunk 2 above)
Comment (jduranf) [reply update.go]: All three cases are covered

Comment (falkerson) [reply update.go]: Sorry, but I couldn't find any related unit tests

File: update.go (hunk 1 above)
Comment (falkerson) [reply update.go]: tests?

File: update.go (hunk 2 above)
Comment (jduranf) [reply update.go]: Added all tests except valid data. In order to perform such test, it's needed a core-metadata service running (or create a mock service), so should be better to do that in another PR.

**Pull request merged**